pulser/sequence.py

```diff
@@ -13,20 +13,70 @@
 # limitations under the License.
 
 from collections import namedtuple
-from collections.abc import Iterable
 import copy
+from functools import wraps
+from itertools import chain
 import warnings
 
 import numpy as np
 
 import pulser
 from pulser.pulse import Pulse
 from pulser.devices import MockDevice
+from pulser.parametrized import Parametrized, Variable
 from pulser._seq_drawer import draw_sequence
 from pulser.utils import validate_duration
 
 # Auxiliary class to store the information in the schedule
 _TimeSlot = namedtuple('_TimeSlot', ['type', 'ti', 'tf', 'targets'])
+# Encodes a sequence building calls
+_Call = namedtuple("_Call", ['name', 'args', 'kwargs'])
+
+
+def _save(func):
```

Comment:
You're absolutely right, we actually don't. I was trying to distinguish calls that you will always execute right away, which are saved always in `_calls`, and the ones who's location (`_calls` or `_to_build_calls`) depend on whether you're actively building the sequence or not (`self._building` is `True` or `False`). 

Once I moved on to the serialization, I realized that the cases where you use `_save` are actually better served with a manual addition to `_calls`, so I got rid of `_save` altogether. I have that ready in another branch, which I rebased from this one. I guess we can either let this be, knowing it will be modified on the next PR, or perhaps I merge the other branch with this one and we turn it into a bigger PR with `serialization` already included? Let me know what you prefer.